modules/electromagnetics/src/kernels/ADFuncCoupledForce.C

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADFuncCoupledForce.h"
+#include "Function.h"
+
+registerMooseObject("ElectromagneticsApp", ADFuncCoupledForce);
+
+InputParameters
+ADFuncCoupledForce::validParams()
+{
+  InputParameters params = ADCoupledForce::validParams();
+  params.addClassDescription(
+      "Kernel representing the contribution of the PDE term $-cfv$, where $c$ and $f$ are constant "
+      "and function coefficients, respectively, and $v$ is a coupled scalar variable.");
+  params.addParam<FunctionName>("func", 1.0, "Function multiplier for the coupled force term.");
+  return params;
+}
+
+ADFuncCoupledForce::ADFuncCoupledForce(const InputParameters & parameters)
+  : ADCoupledForce(parameters), _func(getFunction("func"))
+{
+}
+
+ADReal
+ADFuncCoupledForce::computeQpResidual()
+{
+  return _func.value(_t, _q_point[_qp]) * ADCoupledForce::computeQpResidual();
```

Comment:
I think maybe we should just add a `func` parameter to `ADCoupledForce`. This object seems very general so a good candidate to put its capability in the existing framework object

Reply:
Combined this object with (AD)CoupledForce and turned the coefficient into a functor.

Reply:
Nevermind my previous message for this object. I decided to make the function parameter a material property parameter instead. I can still move this into the main framework as ADMatCoupledForce if you want @lindsayad. 

I've discussed with @GiudGiud how best to use functors here, but it looks like I need to do a bit more work than the prototype code I put together. I think I'll hold that for a future PR.

Reply:
I don't really care much. I leave it to your discretion